Fix issue #5995: [Resolver] Resolver's summary suggests UNRESOLVED due to "no human reviewer"

openhands-agent opened the pull request:
This pull request fixes #5995.

The issue has been successfully resolved. The AI agent has addressed the core problem by modifying the prompt file to ensure the LLM makes independent assessments without deferring to human reviewers. Specifically:

1. The prompt was modified to make the LLM self-reliant in its judgment
2. Instructions were added to prevent waiting for human review
3. Focus was shifted to evaluating concrete changes and their effects
4. Questions were reworded to emphasize direct assessment of changes

This directly addresses the original issue which was that the status overview was incorrectly requiring external human review. The changes to the prompt file will ensure future responses focus on the LLM's own analysis of success based on the changes made, rather than deferring to human reviewers.

A suitable PR message would be:
"Modified the success assessment prompt to ensure independent LLM judgment without human review dependency. The changes focus on concrete assessment of changes and their effects, removing references to external verification requirements. This addresses issue #5966 regarding inappropriate human review requirements in status overviews."

Automatic fix generated by [OpenHands](https://github.com/All-Hands-AI/OpenHands/) 🙌

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:45c84b6-nikolaik   --name openhands-app-45c84b6   docker.all-hands.dev/all-hands-ai/openhands:45c84b6
```

xingyaoww commented:
OpenHands, You should also change other three prompt files related to PR review in `openhands/resolver/prompts/guess_success`

openhands-agent commented:
[OpenHands](https://github.com/All-Hands-AI/OpenHands) started fixing the pr! You can monitor the progress [here](https://github.com/All-Hands-AI/OpenHands/actions/runs/12593002900).

openhands-agent commented:
Overview of Changes:

✓ Successfully Resolved:
- Removed language deferring to human reviewers
- Updated prompts to focus on concrete changes and their effects
- Modified system to make independent assessments
- Maintained consistent language about evidence-based evaluation
- Confirmed issue-success-check.jinja already had appropriate wording

Status: All issues appear to have been fully resolved. The changes directly address the core concern about LLM independence in decision-making, and no remaining issues are evident from the provided information.

neubig commented on openhands/resolver/prompts/guess_success/issue-success-check.jinja:
```diff
@@ -1,15 +1,15 @@
-(1) has the issue been successfully resolved?
-(2) If the issue has been resolved, please provide an explanation of what was done in the PR that can be sent to a human reviewer on github. If the issue has not been resolved, please provide an explanation of why.
+(1) Based on the changes made and their expected impact, has the issue been successfully resolved?
+(2) Provide a clear explanation of what was done in the PR and how it addresses the issue. Focus on the concrete changes and their expected effects, not on the need for external verification.
```
```suggestion
(2) Provide a clear explanation of what was done in the PR and whether it addresses the issue. Focus on the concrete changes and their expected effects, not on the need for external verification.
```
This will probably be more effective if the issue is not resolved.

neubig commented on openhands/resolver/prompts/guess_success/pr-feedback-check.jinja:
```diff
@@ -12,10 +12,10 @@ Files locations:
-(1) has the feedback been successfully incorporated?
-(2) If the feedback has been incorporated, please provide an explanation of what was done that can be sent to a human reviewer on github. If the feedback has not been resolved, please provide an explanation of why.
+(1) Based on the changes made and their expected impact, has the feedback been successfully incorporated?
+(2) Provide a clear explanation of what was done and how it addresses the feedback. Focus on the concrete changes and their expected effects, not on the need for external verification.
```
```suggestion
(2) Provide a clear explanation of what was done and whether it addresses the feedback. Focus on the concrete changes and their expected effects, not on the need for external verification.
```

neubig commented on openhands/resolver/prompts/guess_success/pr-review-check.jinja:
```diff
@@ -9,10 +9,10 @@ PR Review Comments:
-(1) has the feedback been successfully incorporated?
-(2) If the feedback has been incorporated, please provide an explanation of what was done that can be sent to a human reviewer on github. If the feedback has not been resolved, please provide an explanation of why.
+(1) Based on the changes made and their expected impact, has the feedback been successfully incorporated?
+(2) Provide a clear explanation of what was done and how it addresses the feedback. Focus on the concrete changes and their expected effects, not on the need for external verification.
```
```suggestion
(2) Provide a clear explanation of what was done and whether it addresses the feedback. Focus on the concrete changes and their expected effects, not on the need for external verification.
```

neubig commented on openhands/resolver/prompts/guess_success/pr-thread-check.jinja:
```diff
@@ -9,10 +9,10 @@ PR Thread Comments:
-(1) has the feedback been successfully incorporated?
-(2) If the feedback has been incorporated, please provide an explanation of what was done that can be sent to a human reviewer on github. If the feedback has not been resolved, please provide an explanation of why.
+(1) Based on the changes made and their expected impact, has the feedback been successfully incorporated?
+(2) Provide a clear explanation of what was done and how it addresses the feedback. Focus on the concrete changes and their expected effects, not on the need for external verification.
```
```suggestion
(2) Provide a clear explanation of what was done and whether it addresses the feedback. Focus on the concrete changes and their expected effects, not on the need for external verification.
```

neubig reviewed (approved):
LGTM now, thanks!
